clients/input_copy.go

```diff
@@ -23,36 +23,48 @@ const MaxCopyFileDuration = 2 * time.Hour
 const PresignDuration = 24 * time.Hour
 
 type InputCopier interface {
-	CopyInputToS3(requestID string, inputFile, osTransferURL *url.URL) (video.InputVideo, string, error)
+	CopyInputToS3(requestID string, inputFile *url.URL) (video.InputVideo, string, *url.URL, error)
 }
 
 type InputCopy struct {
-	S3    S3
-	Probe video.Prober
+	S3              S3
+	Probe           video.Prober
+	SourceOutputUrl string
 }
 
 // CopyInputToS3 copies the input video to our S3 transfer bucket and probes the file.
-func (s *InputCopy) CopyInputToS3(requestID string, inputFile, osTransferURL *url.URL) (inputVideoProbe video.InputVideo, signedURL string, err error) {
-	if osTransferURL == nil {
-		err = errors.New("osTransferURL was nil")
-		return
-	}
-
-	log.Log(requestID, "Copying input file to S3", "source", inputFile.String(), "dest", osTransferURL.String())
-	size, err := CopyFile(context.Background(), inputFile.String(), osTransferURL.String(), "", requestID)
-	if err != nil {
-		err = fmt.Errorf("error copying input file to S3: %w", err)
-		return
-	}
-	if size <= 0 {
-		err = fmt.Errorf("zero bytes found for source: %s", inputFile)
-		return
-	}
-	log.Log(requestID, "Copied", "bytes", size, "source", inputFile.String(), "dest", osTransferURL.String())
+func (s *InputCopy) CopyInputToS3(requestID string, inputFile *url.URL) (inputVideoProbe video.InputVideo, signedURL string, osTransferURL *url.URL, err error) {
+	if strings.HasSuffix(inputFile.Host, "storage.googleapis.com") && strings.HasPrefix(inputFile.Path, "/directUpload") {
```

Comment:
If someone sent a pre-signed google url to us then that would fail at mist and go into the mediaconvert fallback. Probably fine otherwise, so I guess I could change it to check queryparams are empty and remove the prefix check?

Reply:
Good point, let's leave it for now then and keep it simple